lib/collection/src/collection.rs

```diff
@@ -1340,44 +1345,34 @@ impl Collection {
     }
 
     pub async fn info(&self, shard_selection: Option<ShardId>) -> CollectionResult<CollectionInfo> {
-        let (all_shard_collection_results, mut info) = {
-            let shards_holder = self.shards_holder.read().await;
-
-            let target_shards = shards_holder.target_shard(shard_selection)?;
-
-            let first_shard = *target_shards.first().ok_or_else(|| {
-                CollectionError::service_error(
-                    "There are no shards for selected collection".to_string(),
-                )
-            })?;
-
-            let info = first_shard.info().await?;
-            let info_futures = target_shards.into_iter().skip(1).map(|shard| shard.info());
-
-            (try_join_all(info_futures).await?, info)
-        };
+        let shards_holder = self.shards_holder.read().await;
+        let shards = shards_holder.target_shard(shard_selection)?;
 
-        all_shard_collection_results
+        let mut requests: futures::stream::FuturesUnordered<_> = shards
             .into_iter()
-            .for_each(|shard_info| {
-                info.status = max(info.status, shard_info.status);
-                info.optimizer_status =
-                    max(info.optimizer_status.clone(), shard_info.optimizer_status);
-                info.vectors_count += shard_info.vectors_count;
-                info.indexed_vectors_count += shard_info.indexed_vectors_count;
-                info.points_count += shard_info.points_count;
-                info.segments_count += shard_info.segments_count;
-                for (key, schema) in shard_info.payload_schema {
-                    match info.payload_schema.entry(key) {
-                        Entry::Occupied(o) => {
-                            o.into_mut().points += schema.points;
-                        }
-                        Entry::Vacant(v) => {
-                            v.insert(schema);
-                        }
-                    };
+            // `info` requests received through internal gRPC *always* have `shard_selection`
+            .map(|shard| shard.info(shard_selection.is_some()))
```

Comment:
Missed review of those changes. Looks like there is a problem that creating a flaky test

Reply:
fixed